File: src/main/java/com/j148/backend/attendance/model/Attendance.java
package com.j148.backend.attendance.model;
import com.j148.backend.contractor.model.Contractor;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Attendance {
    @NotNull(message = "Attendance ID cannot be Null")
    @Min(value = 1, message = "Attendance ID cannot be less than 1")
    private Long attendanceId;
    @NotBlank(message = "Time-in is required")
    private LocalDateTime timeIn;
    @NotBlank(message = " Time-out is required")
    private LocalDateTime timeOut;
    private Register register;
    
    private Contractor contractor;

    public enum Register {
        PRESENT, ABSENT, LATE
    }
}

File: hrmsPython/models/Attendance.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum
from typing import Optional

from dataclasses_json import dataclass_json

from models import Contractor


class Register(Enum):
    PRESENT = "PRESENT"
    ABSENT = "ABSENT"
    LATE = "LATE"

@dataclass_json
@dataclass
class Attendance:
    attendanceId: Optional[int] = None
    timeIn: Optional[datetime] = None
    timeOut: Optional[datetime] = None
    register: Optional[Register] = None
    contractor: Optional[Contractor] = None
    contractor: Contractor = None
